src/payment/strategies/paypal/paypal-sdk.ts

```diff
@@ -1,6 +1,12 @@
 export interface PaypalSDK {
     Button: PaypalButton;
     checkout: PaypalExpressCheckout;
+    FUNDING: PaypalFundingTypeList;
+}
+
+export interface PaypalFundingTypeList {
```

Comment:
You might be able to use an enum here and only allow values of this type to go to the `allowed`/`disallowed`.

```ts
// Using a fixed string makes typescript only accept the actual
// enums and not the equivalent string or a number instead.
// Since they are _ambient_ enums (`declare`) they don't get
// compiled, the only act as a definition of the actual value
// in the API.

declare enum TYPES {
    CARD = 'CARD',
    PAYPAL = 'PAYPAL',
}

export interface PaypalFundingTypeList {
	CARD: PaypalFundingTypes.CARD;
	CREDIT: PaypalFundingTypes.CREDIT;
}

export interface PaypalFundingType {
	allowed?: PaypalFundingTypes[];
	disallowed?: PaypalFundingTypes[];
}
```

Reply:
Oh nice, TIL.

Reply:
OK, I played around with this for most of the day and was unable to make the typescript compiler happy. If you're OK with skipping this for now (it sounds like a nice improvement we can do later), then this is more or less ready for review.